pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go

```diff
@@ -59,6 +59,7 @@ func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataS
 			DataSourceInstanceSettings: backend.ToProto().DataSourceInstanceSettings(instanceSettings),
 		},
 		Queries: []*pluginv2.DataQuery{},
+		Headers: query.Headers,
```

Comment:
Are there headers other than `FromAlert` that you're interested in? If not, might be nice to package it up just as that to make it extra explicit

Reply:
It does make it a big ugly in the datasource:

```
var fromAlert bool
	if req.Headers != nil {
		if req.Headers["FromAlert"] == "true" {
			fromAlert = true
		}
	}
```

So perhaps should add a `FromAlert()` method to the request object in the sdk (which would be compatible with this, and basically just does this).

Reply:
And now that I see it that should be && :-P

Reply:
In terms of other headers, I think the idea is that the property is meant to be passed to plugins and was just skipped - but maybe I remember incorrectly and this was intentional - perhaps @marefr knows.

Reply:
Yeah it does look a bit 💩  alright. Having it in the SDK could be an option too, but hopefully wouldn't unnecessarily complicate making changes by splitting the request composition.

Either way though probably a good idea to only pass what we need considering the cruft that gets thrown in headers most of the time - just my 2 cents!